web/app/scripts/filterbar/date-range-directive.js

```diff
@@ -2,54 +2,115 @@
+                        }
+                        if (dtRange.max) {
+                            var jsMax = moment(dtRange.max, moment.ISO_8601).toDate();
+                            scope.max = calendar.formatDate(
+                                scope.calendarOptions.dateFormat,
+                                calendar.fromJSDate(jsMax)
+                            );
+                            $('#dtMaxField').calendarsPicker('setDate', calendar.fromJSDate(jsMax));
+                        }
                         scope.isMinMaxValid();
                     }
                 });
-                scope.$watch('dtMin', function(newMin) {
-                    $('#dtMinField').datepicker('update', newMin);
-                    scope.onDtRangeChange();
-                });
-                scope.$watch('dtMax', function(newMax) {
-                    $('#dtMaxField').datepicker('update', newMax);
-                    scope.onDtRangeChange();
-                });
 
-                // On change of DT value
-                scope.onDtRangeChange = function() {
-                    dtRange.min = scope.dtMin;
-                    dtRange.max = scope.dtMax;
+                /**
+                 * Update the scope and dtRange date variables whenever a new date is selected
+                 * in the calendarPicker. This is fired by the calendar's onSelect action.
+                 * @param date {CDate} The newly selected date, as a CDate.
+                 * @param minOrMax {string} A string specifying 'min' or 'max'.
+                 */
+                function updateDate(date, minOrMax) {
+                    scope[minOrMax] = calendar.formatDate(scope.calendarOptions.dateFormat, date);
+                    var gregCDate = DateLocalization.convertToCalendar(date, 'gregorian', 'en');
+                    var gregISODate = gregCDate._calendar.toJSDate(gregCDate).toJSON();
+                    dtRange[minOrMax] = gregISODate;
                     scope.updateFilter();
-                };
+                }
+
+                /**
+                 * Sets up the calendar picker based on the current language.
+                 */
+                function configureDatePicker() {
+                    var dateConfig = DateLocalization.currentDateFormats();
+                    calendar = $.calendars.instance(dateConfig.calendar, dateConfig.language);
+                    $.calendarsPicker.setDefaults($.calendarsPicker.regionalOptions['']);
+                    scope.calendarOptions = angular.extend({
+                        calendar: calendar,
+                        dateFormat: dateConfig.formats.numeric,
+                        showAnim: '',
+                    }, $.calendarsPicker.regionalOptions[dateConfig.language]);
 
+                    // Make sure that the datepicker hides when the parent dropdown does
+                    elem.on('hide.bs.dropdown', function() {
```

Comment:
@CloudNiner This is related to the visibility issue we've been discussing; without this event listener, the parent drop down with the text fields sometimes disappears without notifying the child calendar element, leaving the calendar floating above the map.

However, the parent dropdown hides itself when the calendar is clicked, so with this fix in place, interacting with the calendar causes both the parent dropdown and the calendar itself to disappear.

The overall issue is that the parent dropdown shouldn't disappear when the user clicks on the child calendar element, but does.